src/main/java/org/terasology/scenario/components/information/ConcatStringComponent.java

```diff
@@ -18,6 +18,9 @@
 import org.terasology.entitySystem.Component;
 import org.terasology.network.Replicate;
 
+/**
+ * Component that denotes two strings to be concatenated

```

Comment:
I also think that we might want to rename ScenarioStringComponent to StringExpressionComponent or so and similarly the other expression components. "Scenario string" could also mean something else.